src/generateTypeFiles.js

```diff
@@ -106,10 +106,20 @@ export const generateTypeFiles = (
     document: DocumentNode,
     options: Options,
 ) => {
-    const generatedDir = path.join(
-        path.dirname(fileName),
-        options.generatedDirectory ?? '__generated__',
-    );
+    const generatedDir = path.isAbsolute(options.generatedDirectory ?? '')
+        ? path.join(
+              options.generatedDirectory ?? '', // The '' is to quiet flow
+              // fileName is absolute here, so we make it relative to cwd
+              // for more reasonable filenames.  We convert leading ..'s
+              // to `__` so this doesn't escape the output directory.
+              path
+                  .relative(process.cwd(), path.dirname(fileName))
```

Comment:
I did it relative to cwd because this is relative to cwd:
```
findGraphqlTagReferences(process.cwd());
```
so the filenames we see in practice are most likely to be under cwd.  In general they're not under schemaFilePath -- at least for webapp they're not -- so the filenames we'd end up with would be really awkward.